File: Python/labs/lab4/task_2_2/AsianCity.py
from City import City

class AsianCity(City):

    def __init__(self, name: str, temperature: float):
        self.name = name
        self.temperature = temperature
        self.hasWeatherWarning = False

    def getName(self):
        return self.name

    def getTemperature(self):
        return self.temperature

    def getTemperatureScale(self):
        return "Celsius"

    def getHasWeatherWarning(self):
        return self.hasWeatherWarning

    def setHasWeatherWarning(self, hasWeatherWarning):
        self.hasWeatherWarning = hasWeatherWarning

File: Cpp/lab07_Adapter/task_2_1/src/asian_city.cpp
#include "asian_city.hpp"


AsianCity::AsianCity(std::string const& name, double temperature)
: name(name), temperature(temperature), hasWeatherWarning(false) {}

std::string const& AsianCity::getName() const
{
  return this->name;
}

double AsianCity::getTemperature() const
{
  return this->temperature;
}

std::string AsianCity::getTemperatureScale() const
{
  return "Celsius";
}

bool AsianCity::getHasWeatherWarning() const
{
  return this->hasWeatherWarning;
}

void AsianCity::setHasWeatherWarning(bool hasWeatherWarning)
{
  this->hasWeatherWarning = hasWeatherWarning;
}
